var/spack/repos/builtin/packages/py-pyarrow/package.py

```diff
@@ -20,6 +20,8 @@ class PyPyarrow(PythonPackage, CudaPackage):
 
     license("Apache-2.0")
 
+    version("16.1.0", sha256="15fbb22ea96d11f0b5768504a3f961edab25eaf4197c341720c4a387f6c60315")
```

Comment:
Is it worth updating `py-numpy` based on `pyproject.toml`?

```
requires = [
    "cython >= 0.29.31",
    # Starting with NumPy 1.25, NumPy is (by default) as far back compatible
    # as oldest-support-numpy was (customizable with a NPY_TARGET_VERSION
    # define).  For older Python versions (where NumPy 1.25 is not yet avaiable)
    # continue using oldest-support-numpy.
    "oldest-supported-numpy>=0.14; python_version<'3.9'",
    "numpy>=1.25; python_version>='3.9'",
    "setuptools_scm",
    "setuptools >= 40.1.0",
    "wheel"
]
```

Reply:
Will grab some time to update this as well later today.

Reply:
Updated in https://github.com/spack/spack/pull/44263/commits/5557ddf5ef2021e132c18ec8511ebb4be0473f69

Reply:
@tldahlgren, I'd love if you could give this a second look to make sure I got the details right.